### DISCUSSION TITLE
how fabric gateway point peer to connect?

### DISCUSSION THREAD

**Discussion by muxijunxi** [hyperledger/fabric-gateway-java#147] [Q&A]
when connect channel,my cli want point ORG1's peer to connect,I have peeraddress and tlsrootcert,but I dont konw how set it in fabric gateway

**Comment by bestbeforetoday** [comment]
First thing is to check whether you are using (or can use) Fabric v2.4+. If so, use the newer [Fabric Gateway client API](https://github.com/hyperledger/fabric-gateway) instead. The Fabric samples demonstrate how to configure TLS using that API:

https://github.com/hyperledger/fabric-samples/blob/c723d812e9bf0ef12ddc0a099805aae94e10d6af/asset-transfer-basic/application-gateway-java/src/main/java/App.java#L77-L84

Using fabric-gateway-java, TLS configuration can be set in the [connection profile](https://github.com/hyperledger/fabric-gateway-java/blob/d3e4c5844bd55b40841ed9a3f3faf9a843a72694/src/test/java/org/hyperledger/fabric/gateway/connection-tls.json).
